test/blackbox-tests/test-cases/github7034.t/run.t

```diff
@@ -0,0 +1,73 @@
+Repro for issue #7034.
+
+The "inner" package enables a warning (69: unused-field) in its source code.
+It builds on its own (with a warning) when lang dune is 3.2 or below:
+  $ cat inner/dune-project
+  (lang dune 3.2)
+  (package (name inner))
+  $ dune build --root=inner
+  Entering directory 'inner'
+  File "inner.ml", line 6, characters 11-18:
+  6 | type t = { x : int }
+                 ^^^^^^^
+  Warning 69 [unused-field]: unused record field x.
+  File "inner.ml", line 6, characters 11-18:
+  6 | type t = { x : int }
+                 ^^^^^^^
+  Warning 69 [unused-field]: unused record field x.
+  Leaving directory 'inner'
+
+
+Note that in versions of lang dune above 3.2 this warning becomes an error:
+  $ cp inner/dune-project inner/dune-project.bak
+  $ sed 's/(lang dune 3.2)/(lang dune 3.3)/' inner/dune-project > inner/dune-project.new
+  $ mv inner/dune-project.new inner/dune-project
+  $ dune build --root=inner
+  Entering directory 'inner'
+  File "inner.ml", line 6, characters 11-18:
+  6 | type t = { x : int }
+                 ^^^^^^^
+  Error (warning 69 [unused-field]): unused record field x.
+  Leaving directory 'inner'
+  [1]
+  $ mv inner/dune-project.bak inner/dune-project
+
+Note that the inner package is symlinked into outer/vendored.
+  $ ls outer/vendored/inner | sort
+  dune
+  dune-project
+  inner.ml
+  inner.mli
+
+Building the outer project works when lang dune is 3.2 or below:
+  $ cat outer/dune-project
+  (lang dune 3.2)
+  (package (name outer))
+  $ dune build --root=outer
+  Entering directory 'outer'
+  File "vendored/inner/inner.ml", line 6, characters 11-18:
+  6 | type t = { x : int }
+                 ^^^^^^^
+  Warning 69 [unused-field]: unused record field x.
+  Leaving directory 'outer'
+
+But when lang dune is 3.3 or higher the warning becomes an error:
+  $ cp outer/dune-project outer/dune-project.bak
+  $ sed 's/(lang dune 3.2)/(lang dune 3.3)/' outer/dune-project > outer/dune-project.new
```

Comment:
Instead of using sed and symlinks, perhaps you can use shell functions? E.g.

```
$ function gen-dune-project() {
> cat >$1/dune-project <<EOF
> (lang dune $2)
> EOF
> }
$ gen-dune-project "outer/" "3.3"
```